File: Array&Hashing/groupAnagram/problem.py
def anagrams(list):
    map = {}
    for i in list:
        result = ''.join(sorted(i))
        
        if result in map:
            map[result].append(i)
        else:
            map[result] = [i]

    
    for group in map.values():
        print(group)
        print()

    return map.values()

strs = ["act","pots","tops","cat","stop","hat"]
results = anagrams(strs)

print(results)

File: Array&Hashing/groupAnagram/problem.js
function groupAnagrams(strs) {

 
//______________________( hash mapping )_________________
  let map = {};

  for (let i = 0; i <= strs.length - 1; i++) {
    let sortedword = strs[i].split("").sort().join("");
     
    if (!hashMap[sortedword]) {
      hashMap[sortedword] = [strs[i]];
    } else {
      
      hashMap[sortedword].push(strs[i]);
    }
  }
  let result = Object.values(hashMap);
  
  return result;
}

console.log(groupAnagrams(["act","pots","tops","cat","stop","hat"]));

